src/Workleap.Extensions.MediatR.Analyzers/SemanticDesignAnalyzer.cs

```diff
@@ -119,7 +150,14 @@ private void AnalyzeOperationInvocation(OperationAnalysisContext context)
                 }
                 else
                 {
-                    context.ReportDiagnostic(HandlersShouldNotCallHandlerRule, operation);
+                    if (this.ImplementsRequestHandlerInterface(containingType))
+                    {
+                        context.ReportDiagnostic(HandlersShouldNotCallHandlerRule, operation);
+                    }
+                    else if (this.ImplementsNotificationHandlerInterface(containingType))
+                    {
+                        context.ReportDiagnostic(NotificationHandlersShouldNotCallHandlerRule, operation);
+                    }
```

Comment:
Maybe just return on line 149 and then the else statement on line 151 can be removed and this block of code can be brought 1 step shallower

Reply:
Done